### PR TITLE
r? ianbicking Fixes #269 Run the 6to5 translation against the addon code

### LINKED ISSUE #269: Use es6 in the addon panel

### PR DESCRIPTION
**Pull request by fzzzy**
Note that you have to generate the xpi inside the addon/dist directory,
now. I will start generating the xpi from the gulpfile next.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+gulp.task("lib-addon", function () {
+  return gulp.src("addon/lib/**/*.{js,jsx}").pipe(to5()).pipe(gulp.dest("addon/dist/lib"));
+});
+
+gulp.task("test-addon", function () {
+  return gulp.src("addon/test/**/*.{js,jsx}").pipe(to5()).pipe(gulp.dest("addon/dist/test"));
+});
+
+gulp.task("static-addon", function () {
```
Comment (ianb) [inline gulpfile.js]: We'll have some SVG shortly (Bram has made some SVG icons I believe)

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+});
+
+gulp.task("javascript-addon", ["data-addon", "lib-addon", "test-addon", "static-addon"], function () {
+  var bundler = browserify({
+    entries: ["./addon/dist/addon-main.js"],
+    debug: true
+  });
+
+  return (function () {
+    return bundler.bundle().pipe(source("bundle.js")).pipe(gulp.dest("addon/dist"));
```
Comment (ianb) [inline gulpfile.js]: Maybe we should name bundles differently, i.e., addon-bundle.js vs website-bundle.js

**Comment by ianb** [comment]
Looks like this still needs a change against the run script too, to run gulp before each cfx invocation and use `--pkgdir`?

**Comment by fzzzy** [comment]
Eventually gulp is going to run the run script, instead of the run script running gulp.

File: gulpfile.js (hunk 1 above)
Comment (fzzzy) [inline gulpfile.js]: Added svg.

**Comment by fzzzy** [comment]
I added code to the gulp script that generates the addon.

**Pull request merged**